src/promptflow/promptflow/_sdk/_telemetry/activity.py

```diff
@@ -172,8 +175,13 @@ def wrapper(self, *args, **kwargs):
             # update activity name according to kwargs.
             _activity_name = update_activity_name(activity_name, kwargs=kwargs)
             with log_activity(logger, _activity_name, activity_type, custom_dimensions):
-                return f(self, *args, **kwargs)
-
+                try:
+                    return f(self, *args, **kwargs)
+                finally:
+                    if _activity_name in HINT_ACTIVITY_NAME:
```

Comment:
shall we start the check and hint for update at the begining of the command?

Reply:
move to beginning